decredplugin/decredplugin.go

```diff
@@ -199,3 +203,173 @@ func DecodeVoteResultsReply(payload []byte) (*VoteResultsReply, error) {
 
 	return &v, nil
 }
+
+// CommentCensor is a journal entry for a censored comment.  The signature and
+// public key are from the admin that censored this comment.
+type CommentCensor struct {
+	Token     string `json:"token"`     // Censorship token
+	CommentID string `json:"commentid"` // Comment ID
+	Signature string `json:"signature"` // Admin Signature of Token+CommentID
+	PublicKey string `json:"publickey"` // Pubkey used for Signature
+}
+
+// CommentCensorReply returns the receipt for the cenosring action. The receipt
+// is the politeaid side signature of CommentCensor.Signature.
+type CommentCensorReply struct {
+	Receipt string `json:"receipt"` // Server signature of the admin Signature
+}
+
+// Comment is the structure that describes the full server side content.  It
+// includes server side meta-data as well. Note that the receipt is the server side
+type Comment struct {
+	// Data generated by client
+	Token     string `json:"token"`     // Censorship token
+	ParentID  string `json:"parentid"`  // Parent comment ID
+	Comment   string `json:"comment"`   // Comment
+	Signature string `json:"signature"` // Client Signature of Token+ParentID+Comment
+	PublicKey string `json:"publickey"` // Pubkey used for Signature
+
+	// Metadata generated by www
+	UserID string `json:"userid"` // User id
+
+	// Metadata generated by decred plugin
+	CommentID string `json:"commentid"` // Comment ID
+	Receipt   string `json:"receipt"`   // Server signature of the client Signature
+	Timestamp int64  `json:"timestamp"` // Received UNIX timestamp
+}
+
+// EncodeComment encodes Comment into a JSON byte slice.
+func EncodeComment(c Comment) ([]byte, error) {
+	b, err := json.Marshal(c)
```

Comment:
Shouldn't this function body just be:

```
return json.Marshal(c)
```

Same for all the `Encode*` functions.